types/rdf-js/index.d.ts

```diff
@@ -472,29 +472,29 @@ export interface DatasetCore<Q extends BaseQuad = Quad> {
     [Symbol.iterator](): Iterator<Q>;
 }
 
-export interface DatasetCoreFactory<Q extends BaseQuad = Quad> {
+export interface DatasetCoreFactory<Q extends BaseQuad = Quad, D extends DatasetCore<Q> = DatasetCore<Q>> {
     /**
      * Returns a new dataset and imports all quads, if given.
      */
-    dataset(quads?: Q[]): DatasetCore<Q>;
+    dataset(quads?: BaseQuad[]): D;
 }
 
-export interface Dataset<Q extends BaseQuad = Quad> extends DatasetCore<Q> {
+export interface Dataset<Q extends BaseQuad = Quad> extends DatasetCore<Q>, Iterable<Q> {
```

Comment:
`Iterable<Q>` would be on `DatasetCore`? (But seems an unrelated change.)

Reply:
Well spotted. I meant to revert that. Thanks.